pkg/test/istio.io/examples/example.go

```diff
@@ -73,46 +84,31 @@ func (example *Example) Delete(namespace string, path string) {
 	example.steps = append(example.steps, newStepFile(namespace, fullPath, true))
 }
 
-// todo: get last script run output
-type testFunc func(t *testing.T)
+type testFunc func(t *testing.T) error
 
 // Exec registers a callback to be invoked synchronously. This is typically used for
 // validation logic to ensure command-lines worked as intended
-func (example *Example) Exec(testFunction testFunc) {
+func (example *Example) Exec(testFunction testFunc) *Example {
 	example.steps = append(example.steps, newStepFunction(testFunction))
+	return example
 }
 
 // getFullPath reads the current gopath from environment variables and appends
 // the specified path to it
-func getFullPath(path string) string {
-	gopath := os.Getenv("GOPATH")
-	return gopath + "/src/" + path
+func getFullPath(scriptPath string) string {
+	return path.Join(env.IstioTop, "/src/"+scriptPath)
 
 }
 
 // Run runs the scripts and capture output
-// Note that this overrides os.Stdout/os.Stderr and is not thread-safe
 func (example *Example) Run() {
-
-	//override stdout and stderr for test. Is there a better way of doing this?
-
-	/*prevStdOut := os.Stdout
-	prevStdErr := os.Stderr
-	defer func() {
-		os.Stdout = prevStdOut
-		os.Stderr = prevStdErr
-	}()*/
-
-	//f, err := os.Create(
-	//os.StdOut =
-
-	scopes.CI.Infof(fmt.Sprintf("Executing test %s (%d steps)", example.name, len(example.steps)))
-
+	scopes.CI.Infof("Executing test %s (%d steps)", example.name, len(example.steps))
 	//create directory if it doesn't exist
-	if _, err := os.Stat(example.name); os.IsNotExist(err) {
-		err := os.Mkdir(example.name, os.ModePerm)
+	if _, err := os.Stat("output"); os.IsNotExist(err) {
+		err := os.Mkdir("output", os.ModePerm)
 		if err != nil {
-			example.t.Fatalf("test framework failed to create directory: %s", err)
+			scopes.CI.Errorf("test framework failed to create directory: %s\n", err)
```

Comment:
Shouldn't this log info go out to the associated testing.T instance?

Reply:
Fixed. Thanks.